src/Core/Usecase/Message/ReceiveMessage.php

```diff
@@ -145,89 +125,129 @@ private function createIncomingMessage($incoming_message, $contact_id, $survey_s
 	 * @param $contact_id
 	 * @param $survey_state_entity
 	 * @param $next_form_attribute
-	 * @return int|$incomingMessageId
-	 * @throws HTTP_Exception_400
+	 * @return int|$outgoingMessageId
 	 */
 	private function createOutgoingMessage($contact_id, $survey_state_entity, $next_form_attribute)
 	{
         // @FIXME: Message type should be configurable per deployment,survey
         $message_type = 'sms';
-        $data_provider = \DataProvider::getEnabledProviderForType($message_type);
```

Comment:
This was breaking phpspec since I added a more thorough spec :(
I moved this handling into data providers themselves so that unassigned messages get sent via the default source